### DISCUSSION TITLE
Unecessary transformation of object spread parameter list?

### DISCUSSION THREAD

**Discussion by csvan** [babel/babel#14661] [Q&A]
Say I have a React (17) component like the following:

```
const AssetItem = ({ title, type, avatar, subtitle, component, secondaryAction, ...rest }) => {
  return (
    <ListItem component={component} {...rest}>
      {avatar || type ? <ListItemIcon>{avatar || <Icon type={type} />}</ListItemIcon> : null}
      <ListItemText primary={title} secondary={subtitle} />
      {secondaryAction}
    </ListItem>
  );
};
```

Running it through Babel (7), I get the following output:

```
const AssetItem = _ref => {
  let {
    title,
    type,
    avatar,
    subtitle,
    component,
    secondaryAction,
    ...rest
  } = _ref;
  return /*#__PURE__*/_jsxs(ListItem, {
    component: component,
    ...rest,
    children: [avatar || type ? /*#__PURE__*/_jsx(_ListItemIcon, {
      children: avatar || /*#__PURE__*/_jsx(Icon, {
        type: type
      })
    }) : null, /*#__PURE__*/_jsx(_ListItemText, {
      primary: title,
      secondary: subtitle
    }), secondaryAction]
  });
};
```

I **don't** want Babel to replace the destructured parameter list with a `_ref` - which it then immediately proceeds to destructure anyway. Is there any way to change this behavior?

My Babel config targets latest 2 major versions of Safari, Chrome, FF and Edge, which all support destructuring (as evident by the output):

```
{
    presets: [
      [
        '@babel/preset-react',
        {
          runtime: 'automatic',
        },
      ],
      '@babel/preset-flow',
      [
        '@babel/preset-env',
        {
          useBuiltIns: 'usage',
          corejs: 3,
          targets: { browsers },
          modules: false,
        },
      ],
    ],
    plugins: [
      'transform-react-remove-prop-types',
      [
        '@babel/plugin-transform-runtime',
        {
          version: '^7.14.0',
        },
      ],
    ],
  };
``

**Comment by JLHwung** [comment]
I can't reproduce this issue on [Babel REPL](https://babel.dev/repl#?browsers=last%202%20safari%20versions%2C%0Alast%202%20firefox%20versions%2C%0Alast%202%20chrome%20versions%2C%0Alast%202%20edge%20versions%2C&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYewdgzgLgBAghCBTKBJKSC2MC8MAUA3jFAJZQA2SANCQJ4AONMAhgG4tQsBOtEArgCMylZqEwNwSMFD5JQYACY86cYGXC0AdDu5JoMAL4BKXAD4YhAFAwYeqP25gCN2zAA8AGVLR0WGOKSYNJQOISBUjKGljpaetCGZq5uluycPDAAPpn0TDAA_B7evhiYqApmhGlc3Fk57uXguUhhUIxI0QD0ZobuncVopY1gFgBcMGD8FBSGyW5ePoNYACpIAB6wDNykmCqt5FTRyArK3HRhAsIHHTDdc7aEx-Cnquqk4LMpfQN-mEm2xgA3FZDICgA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env%2Creact&prettier=false&targets=&version=7.18.4&externalPlugins=&assumptions=%7B%22setClassMethods%22%3Atrue%2C%22noDocumentAll%22%3Atrue%7D)

Can you add `debug: true` to `preset-env` options and attach the enabled Babel plugins?

**Comment by csvan** [comment]
Thanks for the response @JLHwung , this is very strange indeed.

I added the debug option and it gives the following:

```
Based on your code and targets, the corejs3 polyfill did not add any polyfill.
@babel/preset-env: `DEBUG` option

Using targets:
{
  "chrome": "101",
  "edge": "101",
  "firefox": "100",
  "ios": "14",
  "safari": "14"
}

Using modules transform: false

Using plugins:
  proposal-class-static-block { ios, safari }
  proposal-private-property-in-object { ios < 15, safari < 15 }
  proposal-class-properties { ios < 15, safari < 14.1 }
  proposal-private-methods { ios < 15, safari < 15 }
  syntax-numeric-separator
  syntax-nullish-coalescing-operator
  syntax-optional-chaining
  syntax-json-strings
  syntax-optional-catch-binding
  syntax-async-generators
  syntax-object-rest-spread
  proposal-export-namespace-from { ios, safari }
  bugfix/transform-safari-id-destructuring-collision-in-function-expression { ios, safari }
  syntax-dynamic-import
corejs3: `DEBUG` option

Using targets: {
  "chrome": "101",
  "edge": "101",
  "firefox": "100",
  "ios": "14",
  "safari": "14"
}

Using polyfills with `usage-global` method:
```

To the best of my knowledge, I cannot see that any of these should affect the parameter list.

**Comment by JLHwung** [comment]
The parameter transform kicks in because the Safari's parameters compat data is degraded due to this [bug](https://bugs.webkit.org/show_bug.cgi?id=220517). You can specify `preset-env` option `bugfixes: true` (The Babel 8 default) so that Babel only transpiles parameters when it is actually affected by the bug, instead of transpiling unconditionally.

**Reply by nicolo-ribaudo** [reply]
Isn't that bugfix plugin enabled by default (I'm seeing `bugfix/transform-safari-id-destructuring-collision-in-function-expression` I'm the debug output). I'm surprised that it's all transforming arrow functions 🤔

**Reply by JLHwung** [reply]
I can reproduce this issue on [REPL](https://babel.dev/repl#?browsers=safari%2014&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYewdgzgLgBAghCBTKBJKSC2MC8MAUA3jFAJZQA2SANCQJ4AONMAhgG4tQsBOtEArgCMylZqEwNwSMFD5JQYACY86cYGXC0AdDu5JoMAL4BKXAD4YhAFAwYeqP25gCN2zAA8AGVLR0WGOKSYNJQOISBUjKGljpaetCGZq5uluycPDAAPpn0TDAA_B7evhiYqApmhGlc3Fk57uXguUhhUIxI0QD0ZobuncVopY1gFgBcMGD8FBSGyW5ePoNYACpIAB6wDNykmCqt5FTRyArK3HRhAsIHHTDdc7aEx-Cnquqk4LMpfQN-mEm2xgA3FZDICgA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env%2Creact&prettier=false&targets=&version=7.18.5&externalPlugins=&assumptions=%7B%22setClassMethods%22%3Atrue%2C%22noDocumentAll%22%3Atrue%7D) when `bugfixes` is not specified. Guess the REPL is passing `bugfixes: false` in this situation?

**Comment by csvan** [comment]
After blasting node_modules and doing a clean install, it somehow magically works with no other changes :-/ 

Thanks anyway for the help, I will leave this open in case anyone hits a similar issue.
